tests/components/switch_as_x/test_init.py

```diff
@@ -25,3 +28,59 @@ async def test_config_entry_unregistered_uuid(hass: HomeAssistant, target_domain
     await hass.async_block_till_done()
 
     assert len(hass.states.async_all()) == 0
+
+
+@pytest.mark.parametrize("target_domain", ("light",))
+async def test_entity_registry_events(hass: HomeAssistant, target_domain):
+    """Test entity registry events are tracked."""
+    registry = er.async_get(hass)
+    registry_entry = registry.async_get_or_create("switch", "test", "unique")
+    switch_entity_id = registry_entry.entity_id
+    hass.states.async_set(switch_entity_id, "on")
+
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={"entity_id": registry_entry.id, "target_domain": target_domain},
+        title="ABC",
+    )
+
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get(f"{target_domain}.abc").state == "on"
+
+    # Change entity_id
+    new_switch_entity_id = f"{switch_entity_id}_new"
+    registry.async_update_entity(switch_entity_id, new_entity_id=new_switch_entity_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get(f"{target_domain}.abc").state == "unavailable"
```

Comment:
Why does it become unavailable when the tracked entity id changes? I would expect it to still have the state of the tracked entity.

Reply:
When the tracked entity's entity id changes, that entity is first removed from the state machine, and then added again with the new `entity_id`.

The test is maybe a bit too detailed because it does:
1. Change entity_id in entity registry, without setting a state for the new entity_id
2. Check state of the `switch_as_x` entity, which is "unavailable", becuse there's no state for the tracked entity
3. Adds a state for the tracked entity's new `entity_id`
4. Checks that the `switch_as_x` entity is now tracking the new `entity_id` instead of the old one

Reply:
Yeah, normally the entity will receive a state when it's added.